### PR TITLE
Update: yoda onlyEquality option (fixes #2638)

### LINKED ISSUE #2638: Yoda: optionally only apply to equality operator
The premise of the Yoda rule is to prevent accidental assignment instead of comparison due to similarity of `=`, `==`, and `===`. This premise is confirmed in the [issue that initiated the yoda rule in eslint](https://github.com/eslint/eslint/issues/504), [in Wikipedia](http://en.wikipedia.org/wiki/Yoda_conditions), [on Stackoverflow](http://web.archive.org/web/20100715190839/http://stackoverflow.com/questions/2349378/new-programming-jargon-you-coined#answer-2430307) and in [older sources](http://thomas.tuerke.net/on/design/?with=1146181680#msg1146181680). That purpose is fully unrelated to any operators other than the equality operator. I hereby propose making `yoda` rule only apply to equality operator.

The rule as it stands, has caused [questionable exceptions](https://github.com/eslint/eslint/issues/1561) already. Here's another possible one: whether somebody wants yoda always or never is unrelated to the preference towards ordering comparison operators. Somebody might like to always use the smaller-than operator to place the values in the order they lie on the imaginary number axis. Somebody might like to use `yoda: always` to prevent the assignment bug, but dislike the weird look of `2 > a`.

In addition, while the equality operator is commutative, and fixing the warning requires to trivially switch the operands, comparison operators are not commutative, and one also needs to replace the operator. This additional step can be missed resulting in another bug.

Overall, I think it's beneficial for eslint to strive to minimize false positives whenever possible and keep the scope of the rules to a minimum. `yoda` rule is about the equality operator and it should only apply to the equality operator.

I will welcome your input.

### PR DESCRIPTION
**Pull request by denis-sokolov**
PR for #2638.

### CODE REVIEW

File: lib/rules/yoda.js [hunk 1]
```diff
@@ -20,6 +20,15 @@ function isComparisonOperator(operator) {
 }
 
 /**
+ * Determines whether an operator is an equality operator.
+ * @param {String} operator The operator to check.
+ * @returns {boolean} Whether or not it is an equality operator.
+ */
+function isEqualityOperator(operator) {
+    return (/^(==|===)$/).test(operator);
```
Comment (ilyavolodin) [inline lib/rules/yoda.js]: Should this option also apply to `!=` and `!==`? Those are also equality operators.

File: docs/rules/yoda.md [hunk 2]
```diff
@@ -127,6 +127,20 @@ function howLong(arr) {
 }
 ```
 
+### Apply only to equality, but not other operators
```
Comment (ilyavolodin) [inline docs/rules/yoda.md]: I know this md file didn't have options section before. But now that we added an extra option, I think we should reformat it to have options section (like other rules) and show examples. Right now it's not clear whether I can use both `onlyEquality` and `ranges` options, or if I only want to use `onlyEquality` option, should it be third array item or second.

**Comment by ilyavolodin** [comment]
You also need to update `schema` section of the rule with the new option.

File: lib/rules/yoda.js (hunk 1 above)
Comment (denis-sokolov) [inline lib/rules/yoda.js]: I admit there's no consistentcy in the software world, but I don't consider `!=` an "equality operator". It's an "inequality operator".
More importantly, the use case for the option is to split off the operator that can be confused with the assignment operator, and that's `==`, not `!=`.
We can rename the option, if you wish.

File: docs/rules/yoda.md (hunk 2 above)
Comment (denis-sokolov) [inline docs/rules/yoda.md]: I will try.
- [ ] Restructure the options section of the documentation

**Comment by denis-sokolov** [comment]
Updated the schema and the options in the documentation.

**Comment by ilyavolodin** [comment]
LGTM

File: lib/rules/yoda.js (hunk 1 above)
Comment (ilyavolodin) [inline lib/rules/yoda.js]: I don't think that's necessary. We can add `!=` later if there will be enough requests.

**Pull request merged**